### PR TITLE
fix: conversion of environment variables refs does not consider compound cases

### PR DESCRIPTION
**Pull request by warber**
#### What this PR does / Why we need it:
Fixes a bug while converting v1 config with multiple env var references to a v2 config, e.g.:
```yaml
- name: "{{ .Env.alertgroup }} - {{ .Env.application_name}}"
``` 

 produces only a single Environment type parameter for {{alertgroup}} and loses the {{application_name}} variable reference.

The fix contains the code to detect multiple environment variable references and converts them into a compound parameter, e.g.:
```yaml
  - name: "{{ .Env.GOPATH }} - {{ .Env.GOROOT }}"
```
gets converted to:
```yaml
    name:
      format: '{{ .__ENV_GOPATH__ }} - {{ .__ENV_GOROOT__ }}'
      references:
      - __ENV_GOPATH__ 
      - __ENV_GOROOT__ 
      type: compound
    parameters:
      __ENV_GOPATH__ :
        name: GOPATH
        type: environment
      __ENV_GOROOT__ :
        name: GOROOT
        type: environment
```

#### Does this PR introduce a user-facing change?
Multiple env var refs in v1 config yaml files are correctly transformed into compound parameters in the resulting v2 config

### CODE REVIEW

**Review by UnseenWizzard — COMMENTED** [review]
Does this work for environment variables in the config YAML as well? 
In v1 the config YAML is also templated - exclusively for environment variables. 
So the compound env var reference case can happen in the JSON as well as the YAML.

I'm not sure if that takes the same code path.

**Comment by warber** [comment]
yeah I've tried (multiple) env vars in the JSON payload as well. The result is somewhat different of course as it is not converted into a compound variable but I think it is fine:

json:
image

v2 yaml:
image

File: pkg/converter/converter_test.go [hunk 1]
```diff
@@ -54,6 +55,8 @@ func TestConvertParameters(t *testing.T) {
 	listParameterValue := `"GEOLOCATION-41","GEOLOCATION-42","GEOLOCATION-43"`
 	envParameterName := "url"
 	envParameterValue := " {{ .Env.SOME_ENV_VAR }} "
+	compundEnvParameterName := "nickname"
+	compondEnvParameterValue := " {{ .Env.TITLE }} {{ .Env.NICKNAME }} "
```
[the remark this answers is not in the document]
Comment (warber) [reply pkg/converter/converter_test.go:59]: it should not make a difference if n=2 or n=200 but i can certainly add another env var :)

[the remark this answers is not in the document]
Comment (UnseenWizzard) [reply pkg/converter/converter_test.go]: It does not matter until I go and change the converter in a month or two and accidentally break it for n > 2 😉

Thanks!

**Comment by UnseenWizzard** [comment]
> yeah I've tried (multiple) env vars in the JSON payload as well. The result is somewhat different of course as it is not converted into a compound variable but I think it is fine:
> [...]

Perfect thanks! 
As discussed in person, I was thinking about this wrong - works perfectly then.

File: pkg/converter/converter.go [hunk 2]
```diff
@@ -404,8 +404,35 @@ func convertParameters(context *configConvertContext, environment manifest.Envir
...
+						Project:  config.GetProject(),
+						Type:     config.GetType(),
+						ConfigId: config.GetId(),
+					},
+					Property: parameterName,
+				})
+			}
+
+			// convert to compound parameter
+			c, err := compound.New(newName, valueWithPlaceHolders, refs)
```
Comment (UnseenWizzard) [inline pkg/converter/converter.go:430]: For the case of a single variable that is not a compound, couldn't we still exit before adding the compound iff there's only a single param and the `valueWithPlaceholders` is exactly equal to `{{ .Env.[p.Name] }}`?

Comment (warber) [reply pkg/converter/converter.go:430]: yes, but why?

Comment (UnseenWizzard) [reply pkg/converter/converter.go:430]: I think creating a compound where none is needed at all can be confusing to users. 
And if it's an easy fix to not do this with a 3 line early exit if in the right place, I don't see why we should why we should do this only half-way and produce unnecessary complexity.

**Review by UnseenWizzard — APPROVED** [review]
Thanks for addressing all the comments 🎉

**Pull request merged**